web-src/src/components/ShellAuthProvider.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import React, { useEffect, createContext, useState, useContext } from 'react';
+import { ToastQueue } from '@react-spectrum/toast';
+
+import { useApplicationContext } from './ApplicationProvider';
+
+export const ShellAuthContext = createContext({});
+
+export const ShellAuthProvider = ({ children }) => {
```

Comment:
I would suggest merging this provider with the Application Provider, as auth data is not required on its own anywhere in the app. It is only needed to construct the services.

Reply:
I considered separating the Auth context too, as it makes things easier to read and follow in the code.

Reply:
In my mental models, each app has a model but: 
* Models could also be split into classes to isolate the concerns, simplify testing. 
* A single large context that holds all the state for the entire app could lead to unnecessary re-renders.
* Smaller, more focused contexts can be more easily reused across different parts of our app, especially as we change the app based on our learnings. 
* Managing the state of a large, monolithic context can become cumbersome and error-prone.

Reply:
> I considered separating the Auth context too

Separating contexts doesn't necessarily mean separating the providers. The context itself has a modular structure. React does not care if the context object has only one root or many. I would suggest that we at least move the Shell provider above the Application Context.
cc @ddragosd @askayastha22

Reply:
@vtsaplin The Unified Shell config is set in ApplicationProvider and read in ShellAuthProvider. Is moving ShellAuthProvider above ApplicationProvider okay?

Reply:
@vtsaplin let's do this change in a separate PR to unblock the Shell testing